pkg/execution/state/redis_state/redis_state.go

```diff
@@ -463,6 +471,17 @@ func (m mgr) Load(ctx context.Context, runID ulid.ULID) (state.State, error) {
 		return nil, fmt.Errorf("failed to unmarshal event; %w", err)
 	}
 
+	// Load the batch of events
+	cmd = m.r.B().Get().Key(m.kf.Events(ctx, id)).Build()
+	byt, err = m.r.Do(ctx, cmd).AsBytes()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get batch; %w", err)
```

Comment:
We need to check if `err` is a redis nil error;  old function states without batches will error here, but that is actually safe.

This code needs to handle two compat modes:

1. `batches` are not present, use event as the batch
2. Only load batches, and use the first item for the batch

Reply:
Follow up with adding a Version to metadata, in order to mitigate backward incompatibility issues in the executor.